deeplay/list.py

```diff
@@ -37,6 +37,18 @@ def append(self, module: DeeplayModule) -> "LayerList[T]":
             super().append(module)
 
         return self
+    
+    def insert(self, index: int, module: DeeplayModule) -> "LayerList[T]":
+        if not self._has_built:
+            args = list(self._args)
+            args.insert(index, module)
+            self._args = tuple(args)
+            self.__construct__()
+
+        else:
+            super().insert(index, module)
+
+        return self
```

Comment:
I'm assuming the `@before_build` suggestion didn't work because it didn't apply immediately? We can solve that bycalling before_build callbacks immediately as well. I think that would scale better. Then it would simply be

```py
@before_build
def insert(self, index, module):
    return super().insert(index, module)
```